### PR TITLE
Fix deleting / showing `ls` named pipes and other fs objects not explicitly recognized by std::fs

### LINKED ISSUE #1396: Nushell doesn't appear to properly handle named pipes
**Describe the bug**
I've found at least two cases in which a named pipe (from `mkfifo`) isn't being handled correctly in nushell:

1. `ls` will display it as a `symlink`
2. `rm <pipe path>` will fail with "invalid file"

There are likely other issues, but these are the only two I've observed before filing this issue.

**To Reproduce**
1. `mkfifo named.pipe`
2. `ls`
3. `rm named.pipe`

```
error: Invalid file: /Users/gedge/src/github.com/nushell/nushell/named.pipe
- shell:1:13
1 | rm "named.pipe"
  |    ^^^^^^^^^^ invalid file
```

**Expected behavior**
1. `ls` would display the type as "Pipe", or something along those lines.
2. `rm <pipe path>` would remove the pipe.

**Configuration (please complete the following information):**
 - OS: OSX
 - Version 10.15.2

### PR DESCRIPTION
**Pull request by lincis**
This fixes #1396

### CODE REVIEW

**Comment by quebin31** [comment]
It's actually possible to show it's a pipeline by it's file type flag on *nix based systems. IMO I think it would be better to use that flag and show it as `Pipeline` than `Other`

**Comment by jonathandturner** [comment]
Looks like you might need to run `cargo fmt` to pass the formatting check.  You can install it with `rust component add rustfmt`, I believe.

File: crates/nu-cli/src/data/files.rs [hunk 1]
```diff
@@ -2,6 +2,35 @@ use crate::prelude::*;
...
+        {
+            if ft.is_block_device() {
+                file_type = "Block device";
+            } else if ft.is_char_device() {
+                file_type = "Char device";
+            } else if ft.is_fifo() {
+                file_type = "Pipe";
+            } else if ft.is_socket() {
+                file_type = "Socket";
+            }
```
Comment (thegedge) [inline crates/nu-cli/src/data/files.rs]: 🙌 nice work going above and beyond https://github.com/nushell/nushell/issues/1396 and checking for other file types

File: crates/nu-cli/src/data/files.rs [hunk 2]
```diff
@@ -2,6 +2,35 @@ use crate::prelude::*;
 use nu_errors::ShellError;
 use nu_protocol::{TaggedDictBuilder, UntaggedValue, Value};
 
+#[cfg(unix)]
+use std::os::unix::fs::FileTypeExt;
+
+fn get_file_type(md: &std::fs::Metadata) -> &str {
+    let ft = md.file_type();
+    let mut file_type = "Unknown";
```
Comment (thegedge) [inline crates/nu-cli/src/data/files.rs]: In general, if you can avoid mutability, it's a good idea to do so (simpler cognitive model). You can do that here by dropping this variable and writing your if statement this way:

```rust

if ft.is_dir() {
  "Dir"
} ...
} else {
  #[cfg(unix)]
  ...

  "Unknown"
}
```

Still going to land this PR as it is, but wanted to let you know about this 🙂

Comment (lincis) [reply crates/nu-cli/src/data/files.rs]: Thanks, will remember for future, this is my first hands on in rust :)

**Pull request merged**